pkg/operator/cert.go

```diff
@@ -0,0 +1,58 @@
+package operator
+
+import (
+	"crypto/x509"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/service-ca-operator/pkg/operator/util"
+)
+
+// maybeUpdateExpiry updates the expiry of the self-signed CA and returns it if the
+// provided duration is greater than zero. If an error occurs or the duration is
+// zero, the original CA is returned.
+func maybeUpdateExpiry(caConfig *crypto.TLSCertificateConfig, duration time.Duration) (*crypto.TLSCertificateConfig, error) {
+	if duration.Nanoseconds() == 0 {
+		return caConfig, nil
+	}
+	// operator-go's cert generation only accepts an expiry in days. To
+	// ensure support for all validity durations, even those less than a
```

Comment:
Having a flexible initial expiry is required to support validating that cert expiry is handled correctly in the context of a periodic job that takes less than a day to execute (see the description of the [proposed rotation job](https://docs.google.com/document/d/1NB2wUf9e8XScfVM6jFBl8VuLYG6-3uV63eUpqmYE8Ts/edit#heading=h.8rgsj08xt5tp)). Exploratory testing has a similar requirement. If qa or someone else wants to see what happens when a cert expires, I think it makes sense to allow the specification of the appropriate interval (likely in minutes or hours) rather than forcing them to wait a whole day or depend on clock manipulation.

So, there is a requirement. And if this repo is the only one that requires it, I'm not sure why that suggests the effort of updating library-go (especially for an unsupported api).

To your point, though, it is not great UX for a general-purpose option to set the CA expiry in at most hours (since that's the maximum granularity that `Duration` supports). I think that suggests limiting the scope of the option to the initial cert (or manual rotations, since that's the same code path) and changing the name to `ValidityDurationForTesting` to make its use explicit. Forced and automatic rotations would use the default and if there is user demand for varying that duration in the future that would be covered by another option (e.g. `GracefulValidityDuration`) specified in days. WDYT?